pycytominer/tests/test_cyto_utils/test_cells.py

```diff
@@ -293,6 +294,42 @@ def test_load_compartment():
         check_dtype=False,
     )
 
+    # test load_compartment with non-default default_datatype_float
+    # create new SingleCells based on AP
+    float32_loaded_compartment_df = SingleCells(
+        sql_file=TMP_SQLITE_FILE, default_datatype_float=np.float32
+    ).load_compartment(compartment="cells")
+
+    # for uniformly handling metadata types for both dataframes
+    metadata_types = {"ObjectNumber": "int64"}
+
+    # create deep copy of CELLS_DF with manually re-typed float columns as float32
+    cells_df_for_compare = CELLS_DF.copy(deep=True).astype(
+        # cast any float type columns to float32 for expected comparison
+        {
+            colname: np.float32
```

Comment:
is this blacked? I've never seen this format before, but I can read it!

Reply:
I did apply black styling to this, but admittedly the dictionary comprehension mid-block makes things look maybe a bit unusual. Alternatively we could move the dictionary before the `astype` to make this a bit less terse or better matching the reading style of the testing suite. Thoughts?

Reply:
My preference is to split out the dictionary compensation to increase readability. Love the comments though, so please retain these!

Reply:
Thanks, sounds good, will do.